# Bazaar merge directive format 2 (Bazaar 0.90)
# revision_id: hs@ox.cx-20130308172707-zfc1cpf631duoprf
# target_branch: http://svn.twistedmatrix.com/bzr/Twisted/branches\
#   /sslendpoint-certificateoptions-6286/
# testament_sha1: dce9504cd559b1d475a0ac5bcfdecacec9ff6c6c
# timestamp: 2013-03-08 18:36:22 +0100
# base_revision_id: svn-v4:bbbe8e31-12d6-0310-92fd-\
#   ac37d47ddeeb:branches/sslendpoint-certificateoptions-6286:37446
# 
# Begin patch
=== modified file 'twisted/internet/endpoints.py'
--- twisted/internet/endpoints.py	2013-03-06 21:43:58 +0000
+++ twisted/internet/endpoints.py	2013-03-08 17:27:07 +0000
@@ -769,10 +769,12 @@
         kw['method'] = ssl.SSL.SSLv23_METHOD
     certPEM = FilePath(certKey).getContent()
     keyPEM = FilePath(privateKey).getContent()
+    privateCertificate = ssl.PrivateCertificate.loadPEM(certPEM + keyPEM)
     cf = ssl.CertificateOptions(
-        privateKey=ssl.PrivateCertificate.loadPEM(
-            keyPEM + certPEM).privateKey.original,
-        certificate=ssl.Certificate.loadPEM(certPEM).original, **kw)
+        privateKey=privateCertificate.privateKey.original,
+        certificate=privateCertificate.original,
+        **kw
+    )
     return ((int(port), factory, cf),
             {'interface': interface, 'backlog': int(backlog)})
 

# Begin bundle
IyBCYXphYXIgcmV2aXNpb24gYnVuZGxlIHY0CiMKQlpoOTFBWSZTWQfC4aAAAwvfgERQUn//91tK
SgC//9/wUAWvAAAAAAABqTUyZMmQ9JoabTSaMgMCBo0YjHNMTAAmACYAATAAEw5piYAEwATAACYA
AmHNMTAAmACYAATAAEwSSAQAIAiemgGko9TxTynp6myo/SiYIZs4YaSKKpx+9hLQNJ7irhgpAh8Y
WWwAPn1CixlwqPqlKByjqEDxO+ZvPuZjDMzB7f0Kz3iUIeGECDOYueOMA/Z/THw4dwPXxIZF+5wv
8MELygSLTlydrqAfBwEQtFcGzc3bwd3eUT3WwRtEbQDkLMRcIvF1jHsax4VCOoU3OsrEG6bNzdaL
ZtxLZyc50g3kULI7WEO7cOLEI8xCfEA9BFOBgbX6hSEa2FzxfEFchyZVPwFRnsnMUEchhH+dwW9x
lS+BE+pfqMHYmYhwBG0j/YrBGsN5YUSNBWbBH0tH+FoWmsyNMxaxfs3bXGBq2AERGLZY686oZlzq
xNPMcL6mZOCX+hKokWkBHHOsvEcLqpgG0AkIYRWKlJGhVKyqnkSDWQg2ziG41iMdmGRkXmecd8n2
Y9VRYPwsIb3i4hiRvgIt1lLzFWi3mZpYFYaXCMYFpE+AYvCaGJ0vLah+qEC0iaTa6RSZMJCImuQP
EWpD3jdgiT11imahSq0Z0rMJX51BE1mCMi+EwkTtYUEnHB4h2QmC4Kx1RaUnDMIsEyVLioVqLrS6
giAi6giGjx7zHLEUC8T3w2SCYqywoZFd42BXir8yorcCfAsCvQRpjjncFdqePCTi6FlpVbIkD7XG
T1AGwNAoPkSGcKRMeDxwmJjQCsqSm8KDioXA+ZqGG8DWGZcMbXszu3gdh3mgUO4QWqjMhmPyyPVG
oRzYGHqCMC+855E4kzIZK3ygDAVQN3y3Xn87BHKfL6nm/7NWYHRLJebGY8TRoI+LWGR0SYTjyOne
XHZ3cDaR8YuISEea6Dw/QoiPR/o8REoQF4oPLm2fWLDYIxOwuVlWUCucsuHFIhuEWXEz+R/X6+3c
KwsYPcmnmw88RdBBolgI4b+84mhYbxF0OIjeIyR5lYnFBTqs49VohwjwlX3hWCrPHULx8DozkxfT
t+DHUsxF5q0oUPHq7qy0RivQ6GXX1uYZznA5loaJaHVnr1lBEI0Fb1HUIj5GvY4cUq4hjJ5KTh2/
1HhmL2Dgcg6CYWYsRjdzcLIwPYHJbQ5nFGsVgFBf9X4IkEdRpvxvE7icHd5yOfPYJ5YImFrZhcyR
7nYeRrZd4XoJKG5cAN0ni/J1jPQw4Z1hYlJWl96Q1h6HwJiRvBkiKkJwZeS6krOgu0X0cLAxImoC
sXcB9xczBKyR4fMRvNo20GGNDal69SqmAwVDgcDB4nWbmCApaH2PpUZIiXbCoNDcRIBB5sBiYhge
pjpCelgCefgkKEfkXCkKAph8OE4TJMJhfK+roC3EQG8BwoKsVBdaWGisRXAV34EFiNRTUMMJwPRE
vF2gqroC9vcvPzYRE8XARltYIJdBM8RI4mQXCnymn+rIcj7iIi2SifwRBJSiJhXdwnigHaIHQHlw
chqzrAYjJbWTOF1iKLH0CoeVCcC0HiuAcTHH7CodBDxkniYVhAVYdg0DGiFWGYqkqJPEwoMK0eeQ
YcyQnCnmE0mEUFiBAPncYinyAMBAxMWBaMDDhhl/B4ik1aKJ4CP/i7kinChIA+Fw0AA=
